packages/agreement-process/test/testUpgradeAgreement.ts

```diff
@@ -0,0 +1,1273 @@
+        stamps.submission = toStampV1(agreement.stamps.submission);
+      }
+
+      if (agreement.stamps.activation) {
+        stamps.activation = toStampV1(agreement.stamps.activation);
+      }
+
+      if (agreement.stamps.rejection) {
+        stamps.rejection = toStampV1(agreement.stamps.rejection);
+      }
+
+      if (agreement.stamps.suspensionByProducer) {
+        stamps.suspensionByProducer = toStampV1(
+          agreement.stamps.suspensionByProducer
+        );
+      }
+
+      if (agreement.stamps.suspensionByConsumer) {
+        stamps.suspensionByConsumer = toStampV1(
+          agreement.stamps.suspensionByConsumer
+        );
+      }
+
+      if (agreement.stamps.upgrade) {
+        stamps.upgrade = toStampV1(agreement.stamps.upgrade);
+      }
+
+      if (agreement.stamps.archiving) {
+        stamps.archiving = toStampV1(agreement.stamps.archiving);
+      }
+
+      return stamps;
+    }
+
+    async function uploadDocument(path: string, name: string): Promise<void> {
+      await fileManager.storeBytes(
+        config.s3Bucket,
+        config.consumerDocumentsPath,
+        path,
+        name,
+        Buffer.from("large-document-file")
+      );
+      expect(await fileManager.listFiles(config.s3Bucket)).toContainEqual(
+        `${config.consumerDocumentsPath}/${path}/${name}`
+      );
+    }
+
+    it("should succeed with valid Verified and Declared attributes", async () => {
+      const authData = {
+        ...getRandomAuthData(),
+        userRoles: [userRoles.INTERNAL_ROLE],
+      };
+      const tenantId = authData.organizationId.toString();
+
+      const validVerifiedTenantAttribute = getMockVerifiedTenantAttribute(
+        unsafeBrandId<AttributeId>(tenantId)
+      );
+      const validVerifiedEserviceAttribute = getMockEServiceAttribute(
+        validVerifiedTenantAttribute.id
+      );
+
+      const validDeclaredTenantAttribute = getMockDeclaredTenantAttribute(
+        unsafeBrandId<AttributeId>(tenantId)
+      );
+      const validDeclaredEserviceAttribute = getMockEServiceAttribute(
+        validDeclaredTenantAttribute.id
+      );
+
+      const descriptorId = generateId<DescriptorId>();
+      const deprecatedDescriptor = {
+        ...getMockDescriptorPublished(
+          descriptorId,
+          [],
+          [[validDeclaredEserviceAttribute]],
+          [[validVerifiedEserviceAttribute]]
+        ),
+        path: "/deprecatedDescriptor/doc",
+        state: descriptorState.deprecated,
+        version: "1",
+      };
+
+      const publishedDescriptor: Descriptor = {
+        ...getMockDescriptorPublished(descriptorId),
+        version: "2",
+      };
+
+      const tenant = getMockTenant(unsafeBrandId<TenantId>(tenantId), [
+        validDeclaredTenantAttribute,
+        validVerifiedTenantAttribute,
+      ]);
+      const agreementToBeUpgraded: Agreement = {
+        ...getMockAgreement(
+          generateId<EServiceId>(),
+          generateId<TenantId>(),
```

Comment:
This test should fail because here you are saying that the `consumerId` is different from the requester id, but upgrade should succeed only if the requester is the consumer (there is an `assertRequesterIsConsumer`).

It is succeeding because of the internal role I commented above - once you change it to admin, the test should fail and requires this fix

Reply:
Done with [commit-9d1b](https://github.com/pagopa/interop-be-monorepo/pull/392/commits/9d1b01f82eb5329a9d2b21835b4e30967b27e2e3)